### DISCUSSION TITLE
[Question][Hw3] Function definition position

### DISCUSSION THREAD

**Discussion by cskwak99** [prosyslab-classroom/cs348-information-security#534] [Q&A]
Hello,

I have a question about function location in Llvm IR.
Is always callee function definition located above caller function definition? It seems natural, but I saw function declarations are located at the bottom, although some caller function exists above.
How can I understand it?

**Comment by Junseon00** [comment]
Can we see an example?

**Comment by hieudz** [comment]
I think this is common for languages that are compiled (like c++ & ocaml). You don't need to delare a function before use (like in python)

**Comment by cskwak99** [comment]
> Can we see an example?

I mean, when we declare function __dbs_sanitizer__ or __coverage__, its declaration goes below main function definition. Although we use it it above.

**Comment by KihongHeo** [comment]
That does not matter.
If you are not sure, you can compile the LL files and execute the program.

**Comment by cskwak99** [comment]
I checked it execute well but just confused. I think my thought was buried in C language style. Thanks for answers
